mainscripts/Trainer.py

```diff
@@ -292,30 +307,101 @@ def main(args, device_args):
             key_events = io.get_key_events(wnd_name)
             key, chr_key, ctrl_pressed, alt_pressed, shift_pressed = key_events[-1] if len(key_events) > 0 else (0,0,False,False,False)
             
+            #Gan-er comment:
```

Comment:
I do like the comments, having them so people can understand the codebase more is useful.